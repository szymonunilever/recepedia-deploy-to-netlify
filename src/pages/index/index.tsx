import { graphql } from 'gatsby';
import React from 'react';
import Layout from 'src/components/Layout/Layout';
import SEO from 'src/components/Seo/Seo';
import { Text, TagName } from 'src/components/lib/components/Text';

import { RecipeListing } from 'src/components/lib/components/RecipeListing';
import Hero from 'src/components/lib/components/Hero';
import { findPageComponentContent } from 'src/utils';
import { RecipeItem } from 'src/components/lib/components/RecipeListing/partials';

const HomePage = ({ data }: HomePageProps) => {
  const page = data.allPage.edges[0].node;
  const components = page.components.items;
  const recipes = data.allRecipe.nodes;

  return (
    <Layout>
      <SEO title="Recepedia Home" />

      <Text tag={TagName['h1']} text={data.allPage.edges[0].node.title} />
      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'LatestAndGreatest'
          )}
          list={recipes}
        />
      </section>

      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'TopRecipes'
          )}
          list={recipes}
        />
      </section>

      <section>
        <Hero
          content={findPageComponentContent(components, 'Hero')}
          viewType="Image"
        />
      </section>
    </Layout>
  );
};

export default HomePage;

export const pageQuery = graphql`
  {
    allPage(filter: { type: { eq: "Home" } }) {
      edges {
        node {
          ...PageFields
        }
      }
    }
    allRecipe(skip: 10) {
      nodes {
        ...RecipeFields
      }
    }
  }
`;

interface HomePageProps {
  data: {
    allPage: {
      edges: Edge<PageNode>[];
    };
    allRecipe: {
      nodes: RecipeItem[];
    };
  };
}

interface Edge<T> {
  node: T;
}

interface PageNode {
  components: {
    items: {
      name: string;
      content: any;
    }[];
  };
  title: string;
  type: string;
}
