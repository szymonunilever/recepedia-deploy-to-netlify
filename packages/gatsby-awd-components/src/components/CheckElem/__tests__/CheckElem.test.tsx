import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { CheckElem, CheckElemProps } from '../index';
import toJson from 'enzyme-to-json';

describe('<CheckElem />', () => {
  let wrapper: ReactWrapper;
  const props: CheckElemProps = {
    type: 1,
    value: 'checkbox',
  };

  const propsRadio: CheckElemProps = {
    type: 0,
    value: 'radio',
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    wrapper = mount(<CheckElem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render radio', () => {
    wrapper = mount(<CheckElem {...propsRadio} />);
    expect(wrapper.find('input').prop('type')).toBe('radio');
  });
});
