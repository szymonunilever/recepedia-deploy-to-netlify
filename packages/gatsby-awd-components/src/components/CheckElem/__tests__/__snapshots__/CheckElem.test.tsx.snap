// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckElem /> matches the snapshot 1`] = `
<CheckElem
  type={1}
  value="checkbox"
>
  <label
    className="checkable__label checkbox"
  >
    <input
      className="checkable__input"
      type="checkbox"
      value="checkbox"
    />
    <span
      className="checkable__checkelem"
    >
      <span
        className="checkable__checkmark"
      />
    </span>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </label>
</CheckElem>
`;
